src/Modules/Admin/Commands.Admin/GetPowerBIActivityEvents.cs

```diff
@@ -0,0 +1,124 @@
+                this.Logger.ThrowTerminatingError($"{nameof(this.StartDateTime)} is not a valid DateTime.");
+            }
+
+            try
+            {
+                DateTime.Parse(this.EndDateTime);
+            }
+            catch
+            {
+                this.validationError = true;
+                this.Logger.ThrowTerminatingError($"{nameof(this.EndDateTime)} is not a valid DateTime.");
+            }
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.validationError)
+            {
+                return;
+            }
+
+            string formattedStartDateTime = $"'{this.StartDateTime}'";
+            string formattedEndDateTime = $"'{this.EndDateTime}'";
+            string formattedFilter = this.ActivityType;
+            if (!string.IsNullOrEmpty(this.ActivityType))
+            {
+                formattedFilter = $"Activity eq '{this.ActivityType}'";
+            }
+
+            var finalResult = this.ExecuteCmdletHelper(formattedStartDateTime, formattedEndDateTime, formattedFilter);
+            this.LogResult(finalResult);
+        }
+
+        private List<object> ExecuteCmdletHelper(string formattedStartDateTime, string formattedEndDateTime, string formattedFilter)
+        {
+            var finalResult = new List<object>();
+            using (var client = this.CreateClient())
+            {
+                ActivityEventResponse response = client.Admin.GetPowerBIActivityEvents(formattedStartDateTime, formattedEndDateTime, null, formattedFilter);
+                while (response.ContinuationToken != null)
+                {
+                    finalResult = finalResult.Concat(response.ActivityEventEntities).ToList();
+                    string formattedContinuationToken = $"'{WebUtility.UrlDecode(response.ContinuationToken)}'";
+                    response = client.Admin.GetPowerBIActivityEvents(formattedStartDateTime, formattedEndDateTime, formattedContinuationToken, formattedFilter);
+                }
+
+                finalResult = finalResult.Concat(response.ActivityEventEntities).ToList();
+            }
+
+            return finalResult;
+        }
+
+        private void LogResult(List<object> result)
+        {
+            switch (this.ResultType)
+            {
+                case (int)OutputType.JsonString:
+                    {
+                        string jsonRepresentation = JsonConvert.SerializeObject(result);
+                        string indented = JValue.Parse(jsonRepresentation).ToString(Formatting.Indented);
+                        this.Logger.WriteObject(indented, true);
+                    }
+                    break;
+                case (int)OutputType.Objects:
+                    this.Logger.WriteObject(result, true);
+                    break;
+                default:
+                    this.Logger.ThrowTerminatingError($"{nameof(this.ResultType)} is not a valid type. Only 0 or 1 are supported.");
```

Comment:
Change message due to ResultType type change

Reply:
It looks like this is no longer needed. If the user gives any random value for the ResultType, powershell is handling it with a proper message.